### ISSUE TITLE
XunitTestClassRunner - IAsyncLifetime's DisposeAsync and IDisposable's Dispose run in parallel

### ISSUE THREAD

**Issue by campersau** [xunit/xunit#1772]
**Summary:**

When a test class fixture implements both IAsyncLifetime and IDisposable, corresponding Dispose methods are executed in parallel despite the documentation:

```csharp
     /// <summary>
    /// Called when an object is no longer needed. Called just before <see cref="M:System.IDisposable.Dispose" />
    /// if the class also implements that.
    /// </summary>
    Task DisposeAsync();
```

**Source:**
https://github.com/xunit/xunit/blob/1fb6b672768bed98f5e7b6656c1bbb3792ed0530/src/xunit.execution/Sdk/Frameworks/Runners/XunitTestClassRunner.cs#L156-L164

**Workaround**
Just use one or another interface

---

This is the same bug as https://github.com/xunit/xunit/issues/1749 but for `XunitTestClassRunner`
